a10_octavia/controller/worker/tasks/a10_network_tasks.py

```diff
@@ -683,127 +683,123 @@ def __init__(self, *arg, **kwargs):
 
-        if conf_floating_ip:
-            for vr in vrid_list:
-                if vr.subnet_id == subnet.id:
-                    break
-            else:
-                vrid_list.append(
-                    data_models.VRID(
-                        vrid=vrid_value,
-                        project_id=lb_resource.project_id,
-                        vrid_port_id=None,
-                        vrid_floating_ip=None,
-                        subnet_id=subnet.id))
+        if not conf_floating_ip:
+            for vrid in vrid_list:
+                self._delete_vrid_port(vrid.vrid_port_id)
+            vrid_value = prev_vrid_value if prev_vrid_value else vrid_value
+            self._remove_device_vrid_fip(vthunder.partition_name, vrid_value)
+            return []
+
+        vrid_floating_ips = []
+        update_vrid_flag = False
+        vrid_list = self._add_vrid_to_list(vrid_list, subnet, lb_resource.project_id)
+        for vrid in vrid_list:
+            subnet = self.network_driver.get_subnet(vrid.subnet_id)
+            vrid.vrid = vrid_value
             if conf_floating_ip.lower() == 'dhcp':
-                for vrid in vrid_list:
-                    subnet = self.network_driver.get_subnet(vrid.subnet_id)
-                    subnet_ip, subnet_mask = a10_utils.get_net_info_from_cidr(
-                        subnet.cidr)
-                    vrid.vrid = vrid_value
-                    if not a10_utils.check_ip_in_subnet_range(
-                            vrid.vrid_floating_ip, subnet_ip, subnet_mask):
-                        try:
-                            # delete existing port associated to vrid in
-                            # question.
-                            if vrid.vrid_port_id:
-                                self.network_driver.delete_port(
-                                    vrid.vrid_port_id)
-                            fip_obj = self.network_driver.create_port(
-                                subnet.network_id, subnet.id)
-                            self.added_fip_ports.append(fip_obj)
-                            vrid.vrid_floating_ip = fip_obj.fixed_ips[0].ip_address
-                            vrid.vrid_port_id = fip_obj.id
-                            update_vrid_flag = True
-                        except Exception as e:
-                            LOG.error(
-                                "Failed to create neutron port for lb_resource: %s",
-                                lb_resource.id)
-                            raise e
-                    vrid_floating_ips.append(vrid.vrid_floating_ip)
+                subnet_ip, subnet_mask = a10_utils.get_net_info_from_cidr(
+                    subnet.cidr)
+                if not a10_utils.check_ip_in_subnet_range(
+                        vrid.vrid_floating_ip, subnet_ip, subnet_mask):
+                    vrid = self._replace_vrid_port(vrid, subnet, lb_resource)
+                    update_vrid_flag = True
```

Comment:
I think we need add below line here also
vrid_floating_ips.append(vrid.vrid_floating_ip)